### ISSUE TITLE
Choice component accessibility improvements

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#125]
Looking at https://github.com/Yoast/yoast-components/pull/124 there are a couple of a11y possible improvements:

When the choice interface renders a selectbox, I'd say the `<fieldset>` element that wraps the select is not necessary. `<fieldset>` elements should be used to group together logically-related form controls, for example "personal details: name, last name, birthdate" etc. would be input fields wrapped by a fieldset. Instead, a select element is a single form control and not a "group".

When the choice interface renders a group of radio buttons, the `<fieldset>` is perfectly appropriate but then it would also need a `<legend>` element. Group of radio buttons always have a "title" before the group that represents a question or the name of the choice: this title needs to be rendered with a `<legend>` element to be correctly reported by screen readers. For a very clear explanation, see https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/

**Comment by afercia** [comment]
Using the configuration from https://github.com/Yoast/yoast-components/issues/114 for testing purposes, see how the fieldset legend gets repeated by screen readers when moving through the radio buttons using the down and up keys:

![screen shot 2016-12-21 at 16 25 14](https://cloud.githubusercontent.com/assets/1682452/21394821/7349c16e-c79a-11e6-9f56-303122144f5b.png)

This gives context to users about the scope of the options. Without a `<legend>` element instead, there's no "title" screen readers can associate to the options.
